datastream-server/src/main/java/com/linkedin/datastream/server/DatastreamTaskImpl.java

```diff
@@ -436,8 +438,21 @@ public int hashCode() {
   @Override
   public String toString() {
     // toString() is mainly for logging purpose, feel free to modify the content/format
-    return String.format("%s(%s), partitionsV2=%s, partitions=%s, dependencies=%s", getDatastreamTaskName(), _connectorType,
-        String.join(",", _partitionsV2), LogUtils.logNumberArrayInRange(_partitions), _dependencies);
+    // When DatastreamTaskImpl is created using constructor that passes _partitionsV2, _partitions is not populated.
+    // When DatastreamTaskImpl is created using constructor that passes _partitions, _partitionsV2 is also populated.
+    // So, if _partitions is not populated, we can assume that only _partitionsV2 is populated.
+    String partitionsV2FormatLog = String.join(",", _partitionsV2);
+    if (_partitions.size() > 0) {
```

Comment:
can this check be `(if (!StringUtils.isBlank(_partitionsV2))` since only v2 is being referenced within the conditional statement?

Reply:
Thanks for the explanation. ParititonsV2 is topic-partitions or stringified partitions based on low-level or high-level tasks.